cleanlab_studio/internal/util.py

```diff
@@ -50,6 +63,127 @@ def init_dataset_source(
         raise ValueError("Invalid dataset source provided")
 
 
+def apply_corrections_snowpark_df(
+    dataset: Any,
+    cl_cols: Any,
+    id_col: str,
+    label_column: str,
+    keep_excluded: bool,
+) -> Any:
+    from snowflake.snowpark.functions import (
+        when,
+        col,
+        is_null,
+        monotonically_increasing_id,
+    )
+
+    # to use lowercase column names, they need to be wrapped in double quotes to become valid identifiers
+    # for example ("col" should be '"col"' so the engine will process the name as "col")
+    # https://docs.snowflake.com/en/sql-reference/identifiers-syntax
+    label_col = quote(label_column)
+    id_col = quote(id_col)
+    action_col = quote("action")
+    corrected_label_col = quote("corrected_label")
+    cleanlab_final_label_col = quote("__cleanlab_final_label")
+
+    corrected_ds = dataset
+    session = dataset.session
+
+    cl_cols_snowflake = session.create_dataframe(cl_cols)
+
+    if id_col not in corrected_ds.columns:
+        corrected_ds = corrected_ds.withColumn(id_col, monotonically_increasing_id())
+
+    corrected_ds = (
+        cl_cols_snowflake.select([id_col, action_col, corrected_label_col])
+        .join(
+            corrected_ds,
+            on=id_col,
+            how="left",
+        )
+        .withColumn(
+            cleanlab_final_label_col,
+            when(is_null(corrected_label_col), col(label_col)).otherwise(col(corrected_label_col)),
+        )
+        .drop(label_col)
```

Comment:
I believe we should be able to just do `.drop(label_col, corrected_label_col)` here and get rid of the `.select()` at the end of this. Let me know if I'm missing something though.

Reply:
There are several (~15ish) other columns in `cl_cols` which is attained from `download_cleanlab_columns`.

Reply:
At the beginning of this query though, we do `cl_cols_snowflake.select([id_col, action_col, corrected_label_col])` before any of the joins, renaming, coalescing, etc, so we should already have gotten rid of those extraneous columns right?